src/views/Pathways/store/slice.js

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+import fromPairs from 'lodash/fromPairs';
+
+const PathwaySectionSlice = createSlice({
+  name: 'PathwaySection',
+  initialState: {
+    pathwayToView:null,
+    allPathways: {},
+    milestoneToView: null,
+    allMilestones:{},
+  },
+  reducers: {
+    setAllPathways: (state, action) => {
+      state.allPathways = fromPairs( action.payload.map(u => [u.id, u]) );
+    },
+    setAllMilestones: (state, action) => {
+      state.allMilestones = fromPairs( action.payload.map(u => [u.id, u]) );
+    },
+    setPathwayToView: (state, action) => {
+      state.pathwayToView = action.payload
+    },
+    setMilestoneToView: (state, action) => {
+      state.milestoneToView = action.payload
+    },
+    setAddOrEditMilestone: (state, {payload : {milestone, milestoneId}}) => {
```

Comment:
`setAddOrEditMilestone` doesn't make sense as a name. Only addOrEditMilestone makes sense. And something which does an add or edit is called an upsert. Read about an upsert. And what is the point of a an if statement? The code in both places does the same basic thing.